lib/app/widgets/form.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:schulcloud/app/app.dart';
+import '../utils.dart';
+
+Future<bool> showDiscardChangesDialog(BuildContext context) {
+  assert(context != null);
+  final s = context.s;
+
+  return showDialog<bool>(
+    context: context,
+    builder: (context) {
+      return AlertDialog(
+        title: Text(s.app_form_discardChanges),
+        content: Text(s.app_form_discardChanges_message),
+        actions: <Widget>[
+          SecondaryButton(
+            onPressed: () => context.navigator.pop(true),
+            child: Text(s.app_form_discardChanges_discard),
+          ),
+          PrimaryButton(
+            onPressed: () => context.navigator.pop(false),
+            child: Text(s.app_form_discardChanges_keepEditing),
+          ),
+        ],
+      );
+    },
+  );
+}
+
+Future<bool> showConfirmDeleteDialog({
```

Comment:
I think it would be cleaner to make this an extension method on `BuildContext`. That makes it more intuitive to use and we're sure that `context != null`.